poseidon2/src/lib.rs

```diff
@@ -10,14 +10,16 @@ extern crate alloc;
 
 mod diffusion;
 mod matrix;
+mod round_numbers;
 use alloc::vec::Vec;
 
 pub use diffusion::{matmul_internal, DiffusionPermutation};
 use matrix::Poseidon2MEMatrix;
-use p3_field::{AbstractField, PrimeField};
+use p3_field::{AbstractField, PrimeField, PrimeField64};
 use p3_symmetric::{CryptographicPermutation, Permutation};
 use rand::distributions::{Distribution, Standard};
 use rand::Rng;
+pub use round_numbers::poseidon2_round_numbers_128;
```

Comment:
Nit: This doesn't need to be `pub`.

Reply:
I made it pub as I assumed we would want to let external users access the function.

Can be useful if you want to generate an instance of Poseidon2 with the round constants not being random.

Reply:
I think the way to achieve that is to have a new function `Poseidon2::new_from_round_constants()` (or similar) which itself calls `poseidon2_round_numbers_128()`.

Reply:
Sure, I might put that into a new PR.